mypy/typestate.py

```diff
@@ -104,15 +106,16 @@ def is_cached_subtype_check(kind: SubtypeKind, left: Instance, right: Instance)
         if info not in TypeState._subtype_caches:
             return False
         cache = TypeState._subtype_caches[info]
-        if kind not in cache:
+        key = (state.strict_optional, kind)
```

Comment:
`kind` is itself a tuple of bools, why don't just add a one more bool there? (Essentially this will be equivalent to `(state.strict_optional, *kind)`.)